### ISSUE TITLE
[Enhancement] KanBan: Save Order Of Issues on Boards

### ISSUE THREAD

**Issue by 6543** [go-gitea/gitea#11161]
Add a filed `ProjectBoardOrder` or use `Priority` from/to Issue struct.
0 = default
highest number is on the top of the list ...

thoughts how to impement the backend:
```
issue with prio 4 -> 2:
new prio is lower so "Update prio=prio+1 where issue.project_id= .... and prio >= 2 and prio < 4"
issue with prio 3 -> 7:
else "Update prio=prio-1 where issue.project_id= .... and prio > 3 and prio <= 7"
x.ID(issue.id).UpdateCol(prio=newPrio)
```
*Pseudo code don't take it to serious*

Blocked by #8346

**Comment by 6543** [comment]
kanban was not merged jet

**Comment by 6543** [comment]
will be closed by #12506

**Comment by JosephLucas** [comment]
Unfortunately PR #12506 has been closed by @patcito  that "cannot commit to redo the backend" (needed to fulfill reviewer requests). Is there any plan for a sustainable order of issues in the kanban ?
